src/api/include/migraphx/migraphx.hpp

```diff
@@ -979,6 +979,23 @@ struct program : MIGRAPHX_HANDLE_BASE(program)
         return arguments(pout, own{});
     }
 
+// Needed to enable stream sync functionality at compile time with onnxruntime
+// Helps deal with compatability so we dont break execution provider.
+#define MIGRAPHX_STREAM_SYNC
+    template <class Stream>
+    /// Overloaded to allow for execution_environment input
+    arguments run_async(const program_parameters& pparams, Stream s) const
+    {
+        migraphx_arguments_t pout;
+        call(&migraphx_program_run_async,
+             &pout,
+             this->get_handle_ptr(),
+             pparams.get_handle_ptr(),
+             s,
+             get_type_name<std::remove_cv_t<std::remove_pointer_t<Stream>>>().c_str());
```

Comment:
Actually, I dont know if we want to be compatible with C++11 or not, but if we do then we can't use the `_t` overloads.

Reply:
Oh shoot.  I'm assuming anyone as of late would be usings a more recent compiler C++(14/17/20) or are we targeting systems that still use C++11 as a default?

Reply:
Yea I dont if we should target c++11 or c++14, I am not sure what onnxrt uses.

Actually, I just realized this could be simplified by changing the function to(notice the `*` next to `Stream`):

```cpp
    template <class Stream>
    arguments run_async(const program_parameters& pparams, Stream* s) const
```

As that will remove the need for `remove_pointer_t`. I dont think we need `remove_cv_t` either as this needs to be non-const.

Reply:
Done.

Reply:
done